bugbot/rules/perfalert_resolved_regression.py

```diff
@@ -0,0 +1,190 @@
+                            skip_ni = True
+
+                if not skip_ni:
+                    self.extra_ni[bug_id] = bug_history
+
+            self.extra_email_info[bug_id] = bug_history
+
+    def get_needinfo_nicks(self):
+        if not self.extra_ni:
+            return
+
+        def _user_handler(user, data):
+            data[user["name"]] = user.get("nick", "")
+
+        user_emails_to_names = {}
+        BugzillaUser(
+            user_names=[bug_ni["status_author"] for bug_ni in self.extra_ni.values()],
+            include_fields=["nick", "name"],
+            user_handler=_user_handler,
+            user_data=user_emails_to_names,
+        ).wait()
+
+        for bug_id, bug_info in self.extra_ni.items():
+            if bug_info["status_author"] in user_emails_to_names:
+                bug_info["nickname"] = user_emails_to_names[bug_info["status_author"]]
+
+    def comment_handler(self, bug, bug_id, bugs):
+        # Gather all comments to match them with the history after
+        self._bug_comments[bug_id] = bug["comments"]
+
+    def history_handler(self, bug):
+        bug_info = self._bug_history.setdefault(str(bug["id"]), {})
+
+        # Get the last resolution change that was made in this bug
+        for change in bug["history"][::-1]:
+            # Get the most recent resolution change first, this is because
+            # it could have changed since the status was changed and by who
+            if not bug_info.get("resolution"):
+                for specific_change in change["changes"]:
+                    if specific_change["field_name"] == "resolution":
+                        bug_info["resolution"] = specific_change["added"]
+                        bug_info["resolution_previous"] = (
+                            specific_change["removed"].strip() or "---"
+                        )
+                        bug_info["resolution_time"] = change["when"]
+                        break
+
+            if bug_info.get("resolution"):
+                # Find the status that the bug was resolved to, and by who
+                for specific_change in change["changes"]:
+                    if specific_change["field_name"] == "status" and specific_change[
+                        "added"
+                    ] in ("RESOLVED", "REOPENED"):
+                        bug_info["status"] = specific_change["added"]
+                        bug_info["status_author"] = change["who"]
+                        bug_info["status_time"] = change["when"]
+                        break
+
+            if bug_info.get("status"):
+                break
+
+    def gather_bugs_info(self, bugs):
+        Bugzilla(
+            bugids=self.get_list_bugs(bugs),
+            historyhandler=self.history_handler,
+            commenthandler=self.comment_handler,
+            commentdata=bugs,
+            comment_include_fields=["text", "creation_time", "author"],
+        ).get_data().wait()
```

Comment:
We could instead retrieve the user comments and history as part of the initial query by adding `comments` and `history` to `include_fields` in `get_bz_params()`.

If you want to select specific comments fields, you could use `.` to specify that as in the following example: `["comments.text", "comments.creation_time", "comments.author"]`.